influxdb_line_protocol/src/delete_parser.rs

```diff
@@ -29,83 +38,188 @@ pub enum Error {
+                    }) => {
+                        // split this expr into smaller binary if any
+                        let mut exprs = vec![];
+                        let split = Self::split_members(table_name, &expr, &mut exprs);
+                        if !split {
+                            return Err(Error::NotSupportPredicate(predicate.to_string()));
+                        }
+                        Ok(exprs)
+                    }
+                    _ => Err(Error::InvalidSemantics(predicate.to_string())),
+                }
+            }
+        }
+    }
+
+    /// Recursively split all "AND" expressions into smaller ones
+    /// Example: "A AND B AND C" => [A, B, C]
+    /// Return false if not all of them are AND of binary expression of
+    /// "column_name = literal" or "column_name != literal"
+    ///
+    /// The split expressions will be converted into data fusion expressions
+    pub fn split_members(
+        table_name: &str,
+        predicate: &SqlParserExpr,
+        predicates: &mut Vec<Expr>,
+    ) -> bool {
+        // Th below code built to be compatible with
+        //  https://github.com/influxdata/idpe/blob/master/influxdbv2/predicate/parser_test.go
+        match predicate {
+            SqlParserExpr::BinaryOp {
+                left,
+                op: BinaryOperator::And,
+                right,
+            } => {
+                if !Self::split_members(table_name, left, predicates) {
+                    return false;
+                }
+                if !Self::split_members(table_name, right, predicates) {
+                    return false;
+                }
+            }
+            SqlParserExpr::BinaryOp { left, op, right } => {
+                // Verify Operator
+                let op = match op {
+                    BinaryOperator::Eq => Operator::Eq,
+                    BinaryOperator::NotEq => Operator::NotEq,
+                    _ => return false,
+                };
+
+                // verify if left is identifier (column name)
+                let column = match &**left {
+                    SqlParserExpr::Identifier(Ident {
+                        value,
+                        quote_style: _, // all quotes are ignored as done in idpe
+                    }) => Expr::Column(Column {
+                        relation: Some(table_name.to_string()),
+                        name: value.to_string(),
+                    }),
+                    _ => return false, // not a column name
+                };
+
+                // verify if right is a literal
+                let value = match &**right {
+                    SqlParserExpr::Identifier(Ident {
+                        value,
+                        quote_style: _,
+                    }) => Expr::Literal(ScalarValue::Utf8(Some(value.to_string()))),
+                    SqlParserExpr::Value(Value::DoubleQuotedString(value)) => {
+                        Expr::Literal(ScalarValue::Utf8(Some(value.to_string())))
```

Comment:
FWIW you can use `use datafusion::prelude::lit` to write this code more concisely like

```suggestion
                        lit(value.to_string())
```